design/state-channel-committee.md

```diff
@@ -0,0 +1,122 @@
+dispute period all together. Instead it employs a committee that can sign off on
+the closing of a channel. Committee members are kept honest by the collateral
+that they put up.
+
+Here, we'd like to introduce a scheme that improves on BRICK by requiring less
+blockchain interactions and smaller signatures. We simplify payment of fees, and
+provide a mitigation against sybil attacks regarding collateral.
+
+Updates
+-------
+
+Updates to the channel state are only valid when they are signed by all
+participants and committee members. For efficiency, these signatures can be
+combined into a single signature using Schnorr or BLS. Each new update contains
+an incremental sequence number.
+
+Fees
+----
+
+Fees to compensate the committee members for their efforts can be paid by
+increasing the committee member's balances in each channel update.
+
+Unilateral closing
+------------------
+
+Closing a channel by a single participant requires a valid state update, and a
+signed statement by a committee majority that this is the latest state update.
+
+Collateral
+----------
+
+Committee members are only allowed to participate once they've secured
+collateral on-chain.
+
+If they sign a state update, and then are found to have signed off on closing
+the channel on an older state update, then they lose their collateral. This
+collateral is mostly burned to prevent sybil attacks. A small part of the
+collateral is paid to the party that provided proof of wrongdoing. This allows a
+single honest committee member to keep the rest of the committee in check.
+
+Bribing
+-------
+
+To prevent bribing attacks the collateral that is burned when a committee
+majority cheats should be larger than the value of whatever is exchanged in the
+channel.
+
+Withholding
+-----------
+
+To prevent withholding attacks by the committee majority, they should not be
+able to retrieve their collateral when they are still participating in state
+channels.
+
+Note that a committee member is free to initiate the closing of channels that it
+participates in to retrieve its collateral.
+
+Privacy
+-------
+
+To prevent committee members from learning about the contents of intermediate
+state updates, everything in the state may be blinded, except for the balances
+of the committee members (so that they can check the fee), and the sequence
+number (so that they know which state is the latest).
+
+Improvements on the state of the art
+------------------------------------
+
+We improve on the PISA and BRICK protocols by removing the need for extra
+unidirectional payment channels to pay the monitoring service / committee.
+Because we require all state updates to be co-signed by the committee they can
+ensure that fees are paid there.
+
+We improve on the BRICK protocol by allowing signatures from all participants
+and committee members to be combined into a single signature using Schnorr or
+BLS signatures.
+
+We improve on the BRICK protocol by requiring less interaction with the
```

Comment:
- Who would be the submitter for the transaction?
- How/who pays the fees of the on chain transaction?

One possibility might be to randomly select a submitter from the committee and subsequently, split the fees of the transaction evenly across all the participants. 

I've ran into a similar issue when thinking about how to aggregate and submit proofs of storage to the chain and seems to be recurrent in all aggregation based schemes.

Reply:
Those are interesting questions, thanks!

In this scheme it is important that anyone can step out at any moment. Participants should be able to get out when the other party goes offline, and committee members should be able to get out if they want to retrieve their collateral (after a while). Which means that we can't select a specific or random party to submit the transaction.

But it would be unfair to have the costs of the transaction fall to a single party, so perhaps it would be best to pre-pay the closing transaction fee when opening the channel. The participants in the channel (but not the committee) would each pay their share of the closing transaction fee to the contract when they open the channel. Whoever submits the closing transaction can then be reimbursed for the gas costs by the contract.

One complication could be the fluctuating gas costs; the actual costs of closing are not known in advance. Maybe the pre-paid gas costs could be calculated pessimistically, and any funds that remain after paying for gas could be returned to the participants.